testing/MLDB-1935-const-binding.py

```diff
@@ -348,7 +349,84 @@ def test_const_wildcard_var(self):
             ['row1', False],
         ])
 
-    # TODO: function calls, extract, LIKE, Bound Parameters?
+    def test_const_builtin_var(self):
+        res = mldb.query("SELECT __isconst(reshape([1,a,3,4], [2,2])) as isconst FROM ds1 ORDER BY rowName()")
+        self.assertTableResultEquals(res, [
+            ['_rowName', 'isconst',],
+            ['row1', False],
+        ])
+
+        res = mldb.query("SELECT __isconst(now()) as isconst FROM ds1 ORDER BY rowName()")
+        self.assertTableResultEquals(res, [
+            ['_rowName', 'isconst',],
+            ['row1', False],
+        ])
+
+    def test_const_builtin_const(self):
+        res = mldb.query("SELECT __isconst(reshape([1,2,3,4], [2,2])) as isconst FROM ds1 ORDER BY rowName()")
+        self.assertTableResultEquals(res, [
+            ['_rowName', 'isconst',],
+            ['row1', True],
+        ])
+
+    def test_const_userfunction_var(self):
+        mldb.put('/v1/functions/fetch', { 'type': 'fetcher' })
+        res = mldb.query("SELECT __isconst(fetch({url: a})) as isconst FROM ds1 ORDER BY rowName()")
+        self.assertTableResultEquals(res, [
+            ['_rowName', 'isconst',],
+            ['row1', False],
+        ])
+
+        mldb.put('/v1/functions/fetch2', { 'type': 'fetcher', 'deterministic':False })
+        res = mldb.query("SELECT __isconst(fetch2({url: 'itdoesntreallymatter'})) as isconst FROM ds1 ORDER BY rowName()")
+        self.assertTableResultEquals(res, [
+            ['_rowName', 'isconst',],
+            ['row1', False],
+        ])
+
+    def test_const_userfunction_const(self):
+        mldb.put('/v1/functions/fetch', { 'type': 'fetcher' })
+        res = mldb.query("SELECT __isconst(fetch({url: 'itdoesntreallymatter'})) as isconst FROM ds1 ORDER BY rowName()")
+        self.assertTableResultEquals(res, [
+            ['_rowName', 'isconst',],
+            ['row1', True],
+        ])
+
+    def test_const_extract_const(self):
+        res = mldb.query("SELECT __isconst(({2 as x})[x]) as isconst FROM ds1 ORDER BY rowName()")
+        self.assertTableResultEquals(res, [
+            ['_rowName', 'isconst',],
+            ['row1', True],
+        ])
+
+    @unittest.expectedFailure
```

Comment:
Does not work yet unfortunately, because we dont (yet) have the const metadata per column of a rowValueInfo. Thats my next refactor.